AmplifyPlugins/API/Tests/APIHostApp/AWSAPIPluginFunctionalTests/GraphQLConnectionScenario3Tests+Subscribe.swift

```diff
@@ -92,114 +91,95 @@ extension GraphQLConnectionScenario3Tests {
-            valueListener: { event in
-                switch event {
+    func testOnCreateCommentSubscriptionWithModel() async throws {
+        let connectingInvoked = AsyncExpectation(description: "Connection connecting")
+        let connectedInvoked = AsyncExpectation(description: "Connection established")
+        let progressInvoked = AsyncExpectation(description: "progress invoked")
+        let task = try await Amplify.API.subscribe(request: .subscription(of: Comment3.self, type: .onCreate))
+        let subscription = await task.subscription
+        Task {
+            for await subscriptionEvent in subscription {
+                switch subscriptionEvent {
                 case .connection(let state):
                     switch state {
                     case .connecting:
-                        break
+                        await connectingInvoked.fulfill()
                     case .connected:
-                        connectedInvoked.fulfill()
+                        await connectedInvoked.fulfill()
                     case .disconnected:
-                        disconnectedInvoked.fulfill()
+                        break
                     }
                 case .data:
-                    progressInvoked.fulfill()
-                }
-        },
-            completionListener: { event in
-                switch event {
-                case .failure(let error):
-                    XCTFail("Unexpected .failed event: \(error)")
-                case .success:
-                    completedInvoked.fulfill()
+                    await progressInvoked.fulfill()
                 }
-        })
-        XCTAssertNotNil(operation)
-        wait(for: [connectedInvoked], timeout: TestCommonConstants.networkTimeout)
+            }
+        }
+        XCTAssertNotNil(task)
+        await waitForExpectations([connectedInvoked], timeout: TestCommonConstants.networkTimeout)
         let uuid = UUID().uuidString
         let testMethodName = String("\(#function)".dropLast(2))
         let title = testMethodName + "Title"
 
-        guard let createdPost = createPost(id: uuid, title: title) else {
+        guard let createdPost = try await createPost(id: uuid, title: title) else {
             XCTFail("Failed to create post")
             return
         }
 
-        guard createComment(postID: createdPost.id, content: "content") != nil else {
+        guard try await createComment(postID: createdPost.id, content: "content") != nil else {
             XCTFail("Failed to create comment with post")
             return
         }
 
-        wait(for: [progressInvoked], timeout: TestCommonConstants.networkTimeout)
-        operation.cancel()
-        wait(for: [disconnectedInvoked, completedInvoked], timeout: TestCommonConstants.networkTimeout)
-        XCTAssertTrue(operation.isFinished)
+        await waitForExpectations([progressInvoked], timeout: TestCommonConstants.networkTimeout)
+        await task.cancel()
```

Comment:
Seeing warning message : `No 'async' operations occur within 'await' expression`

Reply:
latest PR to add API.subscribe removes this https://github.com/aws-amplify/amplify-ios/pull/2236